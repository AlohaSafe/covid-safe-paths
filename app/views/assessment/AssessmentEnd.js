import React from 'react';
import {
  Dimensions,
  Image,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';

import Colors from '../../constants/colors';
import Fonts from '../../constants/fonts';
import AssessmentButton from './AssessmentButton';

const WIDTH = Dimensions.get('window').width;
const HEIGHT = WIDTH * (300 / 375);

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   ctaAction: () => void;
 *   ctaColor?: string;
 *   ctaTitle: string;
 *   description: ReactNode;
 *   footer: ReactNode;
 *   image: string;
 *   title: string;
 * }>} */
const AssessmentShare = ({
  ctaAction,
  ctaColor,
  ctaTitle,
  description,
  children,
  footer,
  image,
  title,
}) => {
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView style={styles.scrollView}>
        <Image source={image} style={{ width: WIDTH, height: HEIGHT }} />
        <View style={styles.scrollViewContent}>
          <Text
            style={[styles.title, { color: ctaColor ? ctaColor : undefined }]}>
            {title}
          </Text>
          <Text style={styles.description}>{description}</Text>
          {children}
        </View>
      </ScrollView>
      <View style={styles.footer}>
        {footer}
        <AssessmentButton
          color={ctaColor}
          onPress={ctaAction}
          title={ctaTitle}
        />
      </View>
    </SafeAreaView>
  );
};

export default AssessmentShare;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.ASSESSMENT_BACKGROUND,
  },
  scrollView: {
    flex: 1,
  },
  scrollViewContent: {
    padding: 20,
  },
  title: {
    fontFamily: Fonts.primaryBold,
    fontSize: 30,
    marginBottom: 10,
  },
  description: {
    fontFamily: Fonts.primaryRegular,
    fontSize: 18,
    lineHeight: 24,
    marginBottom: 30,
  },
  footer: {
    padding: 20,
  },
});
