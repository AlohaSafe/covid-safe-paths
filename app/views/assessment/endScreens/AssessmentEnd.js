import React from 'react';
import {
  Dimensions,
  Image,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';

import { Typography } from '../../../components/Typography';
import Colors from '../../../constants/colors';
import AssessmentButton from '../AssessmentButton';
import { Colors as AssessmentColors } from '../constants';

const WIDTH = Dimensions.get('window').width;
const CONTAINER_WIDTH = 300 / 375;
const HEIGHT = WIDTH * CONTAINER_WIDTH;

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   ctaAction?: () => void;
 *   ctaColor?: string;
 *   ctaTitle?: string;
 *   description?: ReactNode;
 *   footer?: ReactNode;
 *   image: string;
 *   pretitle?: ReactNode;
 *   title: string;
 * }>} */
const AssessmentEnd = ({
  ctaAction,
  ctaColor,
  ctaTitle,
  description,
  children,
  footer,
  image,
  pretitle,
  title,
}) => {
  return (
    <SafeAreaView style={assessmentStyles.container}>
      <ScrollView style={assessmentStyles.scrollView}>
        <Image source={image} style={{ width: WIDTH, height: HEIGHT }} />
        <View style={assessmentStyles.scrollViewContent}>
          {pretitle}
          <Typography
            surveyFont
            style={[
              assessmentStyles.title,
              { color: ctaColor ? ctaColor : undefined },
            ]}>
            {title}
          </Typography>
          {description && (
            <Typography
              surveyFont
              style={assessmentStyles.description}
              testID='description'>
              {description}
            </Typography>
          )}
          {children}
        </View>
      </ScrollView>
      <View style={assessmentStyles.footer}>
        {footer}
        {ctaTitle && (
          <AssessmentButton
            color={ctaColor}
            onPress={ctaAction}
            title={ctaTitle}
          />
        )}
      </View>
    </SafeAreaView>
  );
};

export default AssessmentEnd;

export const assessmentStyles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: AssessmentColors.BACKGROUND,
  },
  scrollView: {
    flex: 1,
  },
  scrollViewContent: {
    padding: 20,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 28,
    paddingVertical: 15,
  },
  description: {
    fontSize: 20,
    lineHeight: 26,
    marginBottom: 30,
    color: Colors.BLACK,
  },
  footer: {
    padding: 20,
  },
  // this is exported and used elsewhere
  // eslint-disable-next-line react-native/no-unused-styles
  boldBlackText: {
    color: Colors.BLACK,
    fontWeight: 'bold',
    fontSize: 20,
  },
});
