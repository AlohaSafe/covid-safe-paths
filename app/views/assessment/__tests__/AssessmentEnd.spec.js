import { render } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';

import AssessmentEnd from '../AssessmentEnd';

test('base', () => {
  const { asJSON, queryByTestId } = render(
    <AssessmentEnd image={{ uri: 'image.png' }} title='End' />,
  );
  expect(asJSON()).toMatchSnapshot();
  expect(queryByTestId('cta')).toBe(null);
});

test('children', () => {
  const { getByTestId } = render(
    <AssessmentEnd image={{ uri: 'image.png' }} title='End'>
      <View testID='children' />
    </AssessmentEnd>,
  );
  expect(getByTestId('children')).toBeTruthy();
});

test('cta', () => {
  const { getByTestId } = render(
    <AssessmentEnd ctaTitle='Next' image={{ uri: 'image.png' }} title='End' />,
  );
  expect(getByTestId('assessment-button')).toBeTruthy();
});

test('description', () => {
  const { getByTestId } = render(
    <AssessmentEnd
      description='Hello'
      image={{ uri: 'image.png' }}
      title='End'
    />,
  );
  expect(getByTestId('description')).toBeTruthy();
});

test('footer', () => {
  const { getByTestId } = render(
    <AssessmentEnd
      footer={<View testID='footer' />}
      image={{ uri: 'image.png' }}
      title='End'
    />,
  );
  expect(getByTestId('footer')).toBeTruthy();
});

test('pretitle', () => {
  const { getByTestId } = render(
    <AssessmentEnd
      image={{ uri: 'image.png' }}
      pretitle={<View testID='pretitle' />}
      title='End'
    />,
  );
  expect(getByTestId('pretitle')).toBeTruthy();
});
