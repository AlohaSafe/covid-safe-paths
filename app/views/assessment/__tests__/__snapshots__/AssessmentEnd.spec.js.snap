// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#F5F8FC",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "uri": "image.png",
            }
          }
          style={
            Object {
              "height": 600,
              "width": 750,
            }
          }
        />
        <View
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "IBMPlexSans-Bold",
                  "fontSize": 30,
                  "marginBottom": 10,
                },
                Object {
                  "color": undefined,
                },
              ]
            }
          >
            End
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
